power_grid_model_c/power_grid_model/include/power_grid_model/component/appliance.hpp

```diff
@@ -94,17 +94,17 @@ class Appliance : public Base {
         }
         return output;
     }
-    template <bool sym>
-    ApplianceShortCircuitOutput get_sc_output(ComplexValue<sym> const& i) const {
+    ApplianceShortCircuitOutput get_sc_output(ComplexValue<false> const& i) const {
         ApplianceShortCircuitOutput output{};
         static_cast<BaseOutput&>(output) = base_output(true);
-        // TODO(NITISH) convert sym output
-        if constexpr (!sym) {
-            output.i = base_i_ * cabs(i);
-            output.i_angle = arg(i * injection_direction());
-        }
+        output.i = base_i_ * cabs(i);
+        output.i_angle = arg(i * injection_direction());
         return output;
     }
+    ApplianceShortCircuitOutput get_sc_output(ComplexValue<true> const& i) const {
+        ComplexValue<false> iabc{i, i * a2, i * a};
+        return get_sc_output(iabc);
+    }
```

Comment:
any particular reason why you're not using `if constexpr (sym)`? i'm fine with this way, just interested 😬

Reply:
No reason actually. The objective is to convert the ComplexValue<true> to ComplexValue<false> then rest of  the things remain the same.
Can be done with a condition statement too. Should I do it that way?

Reply:
it's fine like this. i prefer the `constexpr` solution because it makes it clear that it's the same function, just with two different implementations. that also makes refactoring easier in the future 

e.g. if we decide to go with `tag` structs rather than boolean `sym` values, which would look something like this

```cpp
template <typename T>
concept tag = std::is_empty<T>;

struct sym_tag {};
struct asym_tag {};

template <typename T>
concept calculation_type = tag<T> && (std::same_as<sym_tag, T> || std::same_as<asym_tag, T>);

template <calculation_type T> constexpr is_sym = false;
template <std::same_as<sym_tag> T> constexpr is_sym = true;
```

the refactoring would only require changing the `template` + `if constexpr` statements

but it doesn't really matter because the code already contains a lot of similar cases